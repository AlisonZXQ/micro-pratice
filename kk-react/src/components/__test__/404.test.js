import React from 'react';
import { shallow } from 'enzyme';
import mountTest from '@tests/shared/mountTest';
import NotFound from '../404';

describe('404', () => {
  mountTest(NotFound);

  it('should snapshot correctly', () => {
    const shallowWrapper = shallow(<NotFound />);
    expect(shallowWrapper).toMatchSnapshot();
  });

  it('should have div', () => {
    const shallowWrapper = shallow(<NotFound />);
    expect(shallowWrapper.find('div').exists()).toBe(true);
  });

  it('should have 4 div tags', () => {
    const shallowWrapper = shallow(<NotFound />);
    expect(shallowWrapper.find('div').length).toBe(4)
  });
})

