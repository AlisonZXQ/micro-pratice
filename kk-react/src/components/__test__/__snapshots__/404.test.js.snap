// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`404 should snapshot correctly 1`] = `
<div
  className="root"
>
  <div
    className="m-error-404 f-cb"
  >
    <div
      className="m-pic f-fl"
    >
      <MyIcon
        className="icon"
        type="icon-404"
      />
    </div>
    <div
      className="m-tip f-fl"
    >
      <h1>
        404
      </h1>
      <h2>
        您访问的页面不存在
      </h2>
    </div>
  </div>
</div>
`;
